dbt-tests-adapter/dbt/tests/adapter/incremental/test_incremental_microbatch.py

```diff
@@ -0,0 +1,84 @@
+import os
+from unittest import mock
+
+import pytest
+from freezegun import freeze_time
+
+from dbt.tests.util import relation_from_name, run_dbt
+
+_input_model_sql = """
+{{ config(materialized='table', event_time='event_time') }}
+select 1 as id, TIMESTAMP '2020-01-01 00:00:00-0' as event_time
+union all
+select 2 as id, TIMESTAMP '2020-01-02 00:00:00-0' as event_time
+union all
+select 3 as id, TIMESTAMP '2020-01-03 00:00:00-0' as event_time
+"""
+
+_microbatch_model_sql = """
+{{ config(materialized='incremental', incremental_strategy='microbatch', unique_key='id', event_time='event_time', batch_size='day') }}
+select * from {{ ref('input_model') }}
+"""
+
+
+class BaseMicrobatch:
+    @pytest.fixture(scope="class")
+    def microbatch_model_sql(self):
+        return _microbatch_model_sql
+
+    @pytest.fixture(scope="class")
+    def input_model_sql(self):
+        return _input_model_sql
+
+    @pytest.fixture(scope="class")
+    def models(self, microbatch_model_sql, input_model_sql):
+        return {
+            "input_model.sql": input_model_sql,
+            "microbatch_model.sql": microbatch_model_sql,
+        }
+
+    def assert_row_count(self, project, relation_name: str, expected_row_count: int):
+        relation = relation_from_name(project.adapter, relation_name)
+        result = project.run_sql(f"select count(*) as num_rows from {relation}", fetch="one")
+
+        if result[0] != expected_row_count:
+            # running show for debugging
```

Comment:
`pytest` will already show the results of the count, e.g. `Assertion error: assert 1 != 2 where 1 = ...`. I would rather not send another query to the database if we don't have to. It also feels odd to not have an assertion in a test during happy path.

Reply:
good call! updated to not send another query, and use an assertion in the happy path